src/actions.cpp

```diff
@@ -432,39 +396,31 @@ void evm_contract::process_filtered_messages(const std::vector<silkworm::Filtere
 
 }
 
-void evm_contract::pushtx( eosio::name miner, const bytes& rlptx ) {
-    LOGTIME("EVM START");
-
-    assert_unfrozen();
+void evm_contract::process_tx(const runtime_config& rc, eosio::name miner, transaction& txn) {
+    LOGTIME("EVM START1");
 
-    eosio::check((get_sender() != get_self()) || (miner == get_self()),
+    auto tx = txn.get_tx();
```

Comment:
This should be `const auto&` to avoid copying.

Reply:
`get_tx` will be a const member function that returns a non-const reference, as we need to call `recover_from` on the transaction, which mutates its internal state.